### ISSUE TITLE
The onReady and onInit are being called again by builders for put controllers

### ISSUE THREAD

**Issue by michael-ottink** [jonataslaw/getx#666]
**Describe the bug**
The onReady and onInit are being called again by builders for put controllers

Note that it is different from https://github.com/jonataslaw/getx/issues/407 where it was just the builder

**To Reproduce**
Make a GetMaterialApp and a home widget and in the constructor of the home widget put a controller then have a builder build that widget in the build method of the home widget. The onReady / onInit are called once more for the builder.

Make sure you use `smartManagement: SmartManagement.onlyBuilder.`

I noticed that putting the builder to `global:false` has impact on the result. It will only print once when the builder is local. But this might be confusing behaviour.

**Expected behavior**
I think the onInit and onReady should only be called once for each controller.

**Flutter Version:**
1.20.3

**Getx Version:**
3.8.0

**Describe on which device you found the bug:**
Pixel 3a Emulator / Huawei p30 pro

**Minimal reproduce code**
```
void main() async {
  runApp(App());
}

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return GetMaterialApp(home: HomePage(), smartManagement: SmartManagement.onlyBuilder,);
  }
}

class HomePage extends StatelessWidget {
  HomePage() {
    Get.put<SomeController>(SomeController());
  }

  @override
  Widget build(BuildContext context) {
    return GetBuilder<SomeController>(
      builder: (SomeController controller) => Text('some content'),
    );
  }
}

class SomeController extends GetxController {
  @override
  void onReady() {
    super.onReady();
    print('onReady');
  }
}
```

outputs

```
[GETX] "SomeController" has been initialized
[GETX] "GetMaterialController" has been initialized
I/flutter ( 7309): onReady
I/flutter ( 7309): onReady
Application finished.
Exited (sigterm)

```

**Comment by yousefak007** [comment]
the same problem :(

**Comment by jonataslaw** [comment]
Fix on 3.13

**Comment by norbertkross** [comment]
problem still persists when ever you call get .put()
